### ISSUE TITLE
I3HiveSplitter fails when running in non-writeable environment (Trac #2165)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2157]
Can be reproduced by running muonl3 using an icerec from /cvmfs

Migrated from https://code.icecube.wisc.edu/ticket/2165
```json
{
    "status": "closed", 
    "changetime": "2019-09-26T09:43:23", 
    "description": "Can be reproduced by running muonl3 using an icerec from /cvmfs", 
    "reporter": "chaack", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1569491003666194", 
    "component": "combo reconstruction", 
    "summary": "I3HiveSplitter fails when running in non-writeable environment", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2018-06-21T16:04:46", 
    "milestone": "", 
    "owner": "kjmeagher", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180621T16:09:05: chaack commented:

> here "fails" means it just gets stuck without any error message.

**Comment by icecube-trac** [comment]
Trac update at 20180621T16:22:36: chaack commented:

> 
> 
> ```text
> eval $(/cvmfs/icecube.opensciencegrid.org/py2-v3.0.1/setup.sh)
> source /cvmfs/icecube.opensciencegrid.org/py2-v3.0.1/RHEL_6_x86_64/metaprojects/icerec/V05-02-00/env-shell.sh
> python /cvmfs/icecube.opensciencegrid.org/py2-v3.0.1/metaprojects/icerec/V05-02-00/level3-filter-muon/python/level3_Master.py -i /data/exp/IceCube/2018/filtered/level2/0101/Run00130484/Level2_IC86.2017_data_Run00130484_Subrun00000000_00000000.i3.zst -g /data/exp/IceCube/2018/filtered/level2/0101/Run00130484/Level2_IC86.2017_data_Run00130484_0101_71_375_GCD.i3.zst -o test.i3.zst --hd5output test.hdf --rootoutput test.root
> ```
>

**Comment by icecube-trac** [comment]
Trac update at 20180621T17:06:13: 
* olivas changed owner from "" to "kjmeagher"
* olivas changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20180621T17:06:37: olivas changed milestone from "" to "Summer Solstice 2018"

**Comment by icecube-trac** [comment]
Trac update at 20180706T18:19:26: 
* olivas commented:

> http://code.icecube.wisc.edu/projects/icecube/changeset/163851/IceCube


* olivas changed status from "assigned" to "closed"
* olivas changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20180706T18:29:06: 
* kjmeagher commented:

> Sadly this was the fix for IceHive not I3HiveSplitter. Although presumably the fix should be similar


* kjmeagher changed resolution from "fixed" to ""
* kjmeagher changed status from "closed" to "reopened"

**Comment by icecube-trac** [comment]
Trac update at 20180716T15:42:13: 
* kjmeagher changed resolution from "" to "fixed"
* kjmeagher commented:

> In changeset 164139:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="164139"
> Use a read only stream to read datafiles to prevent crash if the data files are on cvmfs, as well as switch to $I3_BUILD. fixes #2157
> ```


* kjmeagher changed status from "reopened" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:13: 
* olivas commented:

> Milestone Summer Solstice 2018 deleted


* olivas changed milestone from "Summer Solstice 2018" to ""

**Comment by icecube-trac** [comment]
Trac update at 20190926T09:43:23: jvansanten commented:

> In changeset 175712:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="175712"
> Apply r164139 to trunk
> 
> Fixes #2157 once and for all.
> ```
